### ISSUE TITLE
settings supplied in createIndex have no effect

### ISSUE THREAD

**Issue by davidbkemp** [gphat/wabisabi#25]
When supplying settings to createIndex(), the settings have no effect.  It seems like the code that tacks on the trailing slash (lines 72, 73) cause the body to be lost.  Removing the code that tacks on the trailing slash seems to fix the problem. It is not clear why the trailing slash was added, and so I am reluctant to make a pull request that simply removes that code.

Here is an example:
```
      client.createIndex(name = "foo",
        settings = Some( """{"settings": {"number_of_shards" : 1, "number_of_replicas": 0}}""")
      )
```

When you get the settings for "foo", the number of replicas is still the default value of 1.

**Comment by davidbkemp** [comment]
I have pushed an implementation of getSettings() along with a failing test that exposes the problem with createIndex() to a branch:
https://github.com/gphat/wabisabi/compare/master...davidbkemp:get-settings

**Comment by davidbkemp** [comment]
On a hunch, I tried this rewrite of createIndex() and it fixes the problem.  I still do not know why createIndex() needs the trailing slash, but at least this fix preserves that behaviour in case it is needed:
```
  def createIndex(name: String, settings: Option[String] = None): Future[Response] = {
    val req = url(esURL) / name

    // Do something hinky to get the trailing slash on the URL
    // TODO: Find out if we really need to do this.  The tests pass without it!
    val withTrailingSlash = Req(_.setUrl(req.toRequest.getUrl + "/"))

    // Add the settings if we have any
    val withSettingsIfPresent =
      settings.map({ s => withTrailingSlash.setBody(s.getBytes(StandardCharsets.UTF_8)) }).getOrElse(withTrailingSlash)

    doRequest(withSettingsIfPresent.PUT)
  }
```
 I will create a pull request with this fix in it.

**Comment by gphat** [comment]
Merged and released as 2.0.19. Thanks @davidbkemp !
